runtime/lua/vim/filetype/detect.lua

```diff
@@ -245,6 +245,13 @@ function M.cpp(_, _)
   return vim.g.cynlib_syntax_for_cpp and 'cynlib' or 'cpp'
 end
 
+--- @type vim.filetype.mapfn
+function M.CH_fic(_, _)
+  if not vim.o.fileignorecase then
```

Comment:
I don't think this is applicable. Unlike Vim, Nvim never respects 'fileignorecase' when matching filetype, because of Lua.

Reply:
Do you mean "it _shouldn't_ be applicable? Because you can absolutely make it work if you want to.

Reply:
The problem the patch was trying to solve is that `.h` files are still recognized as `cpp` when `g:c:syntax_for_h` and 'fileignorecase' is set.

Without this patch:
```sh
vim --clean --cmd 'let g:c_syntax_for_h = 1 | set fileignorecase' file.h
```
set `cpp` filetype, while
```sh
nvim --clean --cmd 'let g:c_syntax_for_h = 1 | set fileignorecase' file.h
```
already sets `c` filetype.

Reply:
And, with this patch and 'fileignorecase' but without `g:c_syntax_for_h`, Vim set `c` filetype for `.C` files and `cpp` filetype for `.H` files, while Nvim set no filetype at all for `.C` and `.H` files.

Reply:
OK, I'll leave this one for you to handle then.

Reply:
TBH I don't know what to do either...